index.mjs

```diff
@@ -0,0 +1,452 @@
+//. # Warp State
+//.
+//. [![Build Status](https://travis-ci.com/wearereasonablepeople/warp-state.svg?branch=master)](https://travis-ci.com/wearereasonablepeople/warp-state)
+//. [![Coverage Status](https://coveralls.io/repos/github/wearereasonablepeople/warp-state/badge.svg?branch=master&t=Bckm7f)](https://coveralls.io/github/wearereasonablepeople/warp-state?branch=master)
+//. [![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+//.
+//. A state monad implementation compliant to [Fantasy Land][1]
+//. inspired by [fantasy-states][2].
+
+import Z from 'sanctuary-type-classes';
+
+// compose :: (b -> c, a -> b) -> (a -> c)
+export function compose(f, g) {
+  return function(v) {
+    return f (g (v));
+  };
+}
+
+// constant :: a -> b -> a
+export function constant(v) {
+  return function() {
+    return v;
+  };
+}
+
+// next :: a -> c
+function next(v) {
+  return {done: false, value: v};
+}
+
+// done :: b -> c
+function done(v) {
+  return {done: true, value: v};
+}
+
+export function State(run) {
+  if (!(this instanceof State)) return new State (run);
+  this.run = run;
+}
+
+//. ## State
+//.
+//# fantasy-land/of :: a -> State s a
+//.
+//. Fantasy Land compliant implementation of Of.
+//.
+//. ```js
+//.   evalState (null) (Z.of (State, 1)); // 1
```

Comment:
Why the indentation?

Reply:
You could format this as a doctest, in case they will be supported in mjs files in the future:

```js
> evalState (null) (Z.of (State, 1))
1
```